Bio/PDB/Selection.py

```diff
@@ -74,7 +74,7 @@ def unfold_entities(entity_list, target_level):
     else:  # we're going up, e.g. A->S
         for i in range(level_index, target_index):
             # find unique parents
-            entity_list = {entity.get_parent() for entity in entity_list}
+            entity_list = dict.fromkeys([entity.get_parent() for entity in entity_list])
```

Comment:
Yes. Without this change, the unit-tests I added fail.

Reply:
Ah. You're changing the datatype from a set to dict (neither of which is a list so the variable name is very misleading), mainly to go from unordered to ordered.

```pycon
>>> entity_list = [(1,"alpha"), (2,"beta"),(3,"gamma"), (26,"omega"), (4,"delta")]
>>> {entity[1] for entity in entity_list}
set(['alpha', 'beta', 'omega', 'gamma', 'delta'])
>>> dict.fromkeys([entity[1] for entity in entity_list])
{'alpha': None, 'beta': None, 'omega': None, 'gamma': None, 'delta': None}
>>> {entity[1]:None for entity in entity_list}
{'alpha': None, 'beta': None, 'omega': None, 'gamma': None, 'delta': None}
```

Style wise I'd use the dictionary comprehension here, and I think this needs an explanatory comment:

```python
# Need a fast ordered data structure (so not a set)
entity_list = {entity.get_parent():None for entity in entity_list}
```

I find myself asking why isn't this just a list? Perhaps it was a list but a set (with hashing) was more performant (but lost the order)? If so, a dict makes sense.

Reply:
Yes, the set was used initially as a Python idiom to remove duplicate entries in the list, but it removed the order. I changed it to a different Python idiom using `dict.from_keys` which could also be a dict-comprehension. Neither the set, nor the dict were converted back to lists because no list-specific functionality is required and it's converted to a list before returning, so it would be sensible to rename `entity_list` to something more generic and less type-specific.

Given this discussion, would like me to use a dictionary comprehension?

Reply:
+1 on renaming the internal variable ``entity_list``

+1 on using a dict comprehension

But I am happy to defer to @JoaoRodrigues if he is able to comment.

Reply:
I have made the proposed changes.